overlord/devicestate/devicemgr.go

```diff
@@ -1353,3 +1356,11 @@ func (scb storeContextBackend) SignDeviceSessionRequest(serial *asserts.Serial,
 func (m *DeviceManager) StoreContextBackend() storecontext.Backend {
 	return storeContextBackend{m}
 }
+
+func hasFDESetupHook(bootWith *boot.BootableSet) bool {
```

Comment:
now I understand my confusion, what's the plan to check for this at reseal or other places? maybe we don't need to? this approach only works during MakeBootable ? 

if this were a method on DeviceManager we could look at state and the actual kernel, anway the install mode kernel and the kernel we install are the same

Reply:
For the current ARM systems we need to support we need no reseal support AFAICT. I suppose we will need to add a point to BootableSet (or kernel *snap.Info) to TrustedAssetsUpdateObserver so that we can pass that to to resealKeyToModeenv(). 

Alternatively I suppose we could store the information if we use hooks in modeenv (nor sure if that makes sense?)

Reply:
I will add a TODO once we have a direction for how to do this in the future

Reply:
I think looking at the current active kernel is still easiest overall (If not in the detail here), also because anyway we can't really naturally call hooks of a non-installed kernel.